lib/tasks/retire.rake

```diff
@@ -1,36 +1,35 @@
 namespace :retire do
-  desc "Unpublish, redirect and remove from the search index an identified smart answer"
-  task :unpublish_redirect_remove_from_search, %i[content_id base_path destination] => :environment do |_, args|
+  desc "Unpublish a content item with a redirect"
+  task :unpublish_redirect, %i[content_id destination] => :environment do |_, args|
     raise "Missing content_id parameter" if args.content_id.blank?
-    raise "Missing base_path parameter" if args.base_path.blank?
     raise "Missing destination parameter" if args.destination.blank?
 
-    content_item_publisher = ContentItemPublisher.new
-
-    content_item_publisher.unpublish_with_redirect(
-      args.content_id, args.base_path, args.destination
-    )
+    Services.publishing_api.unpublish(args.content_id,
+                                      type: "redirect",
+                                      alternative_path: args.destination)
```

Comment:
Do we need to preserve this behaviour?

https://github.com/alphagov/smart-answers/pull/4429/commits/b66db24a7607c91f7c3c225634cefabff625c928#diff-8cef3e0c497f0b3bd493ed82777c86d2L35

Reply:
Oh shoot! I've probably messed that up - good catch.

Reply:
I'm still wondering: does this conflict with the instructions here: https://github.com/alphagov/smart-answers/pull/4429/files#diff-5450b7bbb07ec3d9169cb6e9fa889847R35? It looks like this is something we've added, compared to [the original](https://github.com/alphagov/smart-answers/pull/4429/files#diff-5450b7bbb07ec3d9169cb6e9fa889847L33), which only suggested running the task once.

Reply:
With the previous approach you should still have done it for both, however it's not clearly signposted: https://github.com/alphagov/smart-answers/pull/4429/files#diff-5450b7bbb07ec3d9169cb6e9fa889847L5-L6 I imagine it's often been omitted as people wouldn't anticipate visits to `/<smart-answer>/y`

Even with a prefix on the `/<smart-answer>` any visits to /<smart-answer>/y would go to that route as prefix only applies when there aren't routes with more matching segments.

Reply:
> prefix only applies when there aren't routes with more matching segments

That's good to know.